src/store/modules/resetpassword/reducer/index.js

```diff
@@ -0,0 +1,31 @@
+import actionTypes from '../actionTypes';
+
+const {
+  PASSWORD_RESET_PENDING,
+  PASSWORD_RESET_SUCCESS,
+  PASSWORD_RESET_FAILED
+} = actionTypes;
+
+const initialState = {
+  isPending: false,
+  isSuccess: false,
+  error: null
+};
+
+const resetPassword = (state = initialState, action) => {
+  switch (action.type) {
+    case PASSWORD_RESET_PENDING:
+      return Object.assign({}, state, { isPending: true });
```

Comment:
To avoid probable bugs in future, rather than hardcoding the new changes to the state. It could come as the action that was dispatched since such action would already have a payload.
e,g,
```
return Object.assign({}, state, ...payload)
```

OR 

```
return {...state, ...payload} 
```

Reply:
Thanks for the feedback. I think this is a matter of choice. Object.assign obeys the third principle of redux which states that reducers should be pure and avoid mutation.

Reply:
This has nothing to do with Object.assign but hardcoding for e.g. `...payload` rather than `{ isPending: true }